tests/test_subdirectory.py

```diff
@@ -152,7 +152,18 @@ def test_update_subdirectory_from_root_path(tmp_path_factory):
     assert (dst / "subfolder" / "file1").read_text() == "version 2\nhello\na1\nbye\n"
 
 
-def test_new_version_uses_subdirectory(tmp_path_factory):
+@pytest.mark.parametrize(
+    "conflict, readme",
+    [
+        ("rej", "upstream version 2\n"),
+        (
+            "inline",
+            "<<<<<<< modified\ndownstream version 1\n"
+            "=======\nupstream version 2\n>>>>>>> new upstream\n",
+        ),
+    ],
+)
+def test_new_version_uses_subdirectory(conflict, tmp_path_factory, readme):
```

Comment:
Expanded this into two test cases, one for the current (`rej`) behavior and one for the new (`inline`) behavior.